src/hal/utils/halcmd_main.c

```diff
@@ -473,15 +473,17 @@ void halcmd_output(const char *format, ...) {
 
 void halcmd_warning(const char *format, ...) {
     va_list ap;
-    fprintf(stderr, "%s:%d: Warning: ", halcmd_get_filename(), halcmd_get_linenumber());
+    if (*halcmd_get_filename() != '\0')
+        fprintf(stderr, "%s:%d: Warning: ", halcmd_get_filename(), halcmd_get_linenumber());
     va_start(ap, format);
     vfprintf(stderr, format, ap);
     va_end(ap);
 }
 
 void halcmd_error(const char *format, ...) {
     va_list ap;
-    fprintf(stderr, "%s:%d: ", halcmd_get_filename(), halcmd_get_linenumber());
+    if (*halcmd_get_filename() != '\0')
+        fprintf(stderr, "%s:%d: ", halcmd_get_filename(), halcmd_get_linenumber());
```

Comment:
and here you check for an empty string. does this mean that "halcmd invalid command" will no longer produce a diagnostic message, but will rather print nothing?

Reply:
In case of stdin and commandline the string is set to empty. In these cases the error message will not include the filename. So the output will only contains the error message that is output with
`   vfprintf(stdout, format, ap);`
in line 488